saleor/product/migrations/0112_auto_20200129_0050.py

```diff
@@ -1,49 +1,64 @@
 # Generated by Django 2.2.9 on 2020-01-29 06:50
 
-from django.db import migrations, models
+from collections import defaultdict
+
+from django.db import migrations, models, transaction
+from django.db.utils import IntegrityError
 from django.utils.text import slugify
 
 
 def create_unique_slugs_for_producttypes(apps, schema_editor):
     ProductType = apps.get_model("product", "ProductType")
 
-    for product_type in ProductType.objects.all():
-        if not product_type.slug:
-            product_type.slug = generate_unique_slug(product_type)
-            product_type.save(update_fields=["slug"])
+    for product_type in ProductType.objects.filter(slug__isnull=True):
+        slug = slugify(product_type.name)
+        extension = 1
+        slug_value = slug
+        while True:
+            try:
+                with transaction.atomic():
+                    product_type.slug = slug_value
+                    product_type.save(update_fields=["slug"])
```

Comment:
This transaction is not needed

Reply:
I need transcation somewhere, otherwise I get an error when running mutation.

Reply:
I changed soultion.